### ISSUE TITLE
devices stay busy even if status-wda or status-adb fail by timeout

### ISSUE THREAD

**Issue by vdelendik** [zebrunner/mcloud-grid#103]
```
21:11:33.049 WARN [MobileRemoteProxy.getNewSession] - 00008110-001C5826148A801E is not ready for a session. /status-wda error: {"value":{"error":"unknown error","message":"An unknown server-side error occurred while processing the command. Original error: Error for sending of WDA status http call. See appium logs for details[[[--udid 00008110-001C5826148A801E --name iPhone_13_Pro_Max --sessionId a4999aac-0b49-4a2e-8595-17009f241202]]]","stacktrace":"UnknownError: An unknown server-side error occurred while processing the command. Original error: Error for sending of WDA status http call. See appium logs for details[[[--udid 00008110-001C5826148A801E --name iPhone_13_Pro_Max --sessionId a4999aac-0b49-4a2e-8595-17009f241202]]]\n    at getResponseForW3CError (/usr/lib/node_modules/appium/node_modules/@appium/base-driver/lib/protocol/errors.js:1073:9)\n    at asyncHandler (/usr/lib/node_modules/appium/node_modules/@appium/base-driver/lib/protocol/protocol.js:494:61)"}}
21:11:34.528 ERROR [HttpClientApache.execute] - Read timed out
java.net.SocketTimeoutException: Read timed out
        at java.base/java.net.SocketInputStream.socketRead0(Native Method)
        at java.base/java.net.SocketInputStream.socketRead(SocketInputStream.java:115)
        at java.base/java.net.SocketInputStream.read(SocketInputStream.java:168)
        at java.base/java.net.SocketInputStream.read(SocketInputStream.java:140)
        at org.apache.http.impl.io.SessionInputBufferImpl.streamRead(SessionInputBufferImpl.java:137)
        at org.apache.http.impl.io.SessionInputBufferImpl.fillBuffer(SessionInputBufferImpl.java:153)
        at org.apache.http.impl.io.SessionInputBufferImpl.readLine(SessionInputBufferImpl.java:280)
        at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:138)
        at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:56)
        at org.apache.http.impl.io.AbstractMessageParser.parse(AbstractMessageParser.java:259)
        at org.apache.http.impl.DefaultBHttpClientConnection.receiveResponseHeader(DefaultBHttpClientConnection.java:163)
        at org.apache.http.impl.conn.CPoolProxy.receiveResponseHeader(CPoolProxy.java:157)
        at org.apache.http.protocol.HttpRequestExecutor.doReceiveResponse(HttpRequestExecutor.java:273)
        at org.apache.http.protocol.HttpRequestExecutor.execute(HttpRequestExecutor.java:125)
        at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:272)
        at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)
        at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
        at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
        at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108)
        at com.zebrunner.mcloud.grid.util.HttpClientApache.execute(HttpClientApache.java:110)
        at com.zebrunner.mcloud.grid.util.HttpClientApache.get(HttpClientApache.java:73)
        at com.zebrunner.mcloud.grid.MobileRemoteProxy.getNewSession(MobileRemoteProxy.java:122)
        at org.openqa.grid.internal.ProxySet.lambda$getNewSession$3(ProxySet.java:130)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
        at java.base/java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1632)
        at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:127)
        at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:502)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:488)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
        at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:543)
        at org.openqa.grid.internal.ProxySet.getNewSession(ProxySet.java:132)
        at org.openqa.grid.internal.DefaultGridRegistry.takeRequestHandler(DefaultGridRegistry.java:251)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:176)
        at java.base/java.util.stream.SortedOps$SizedRefSortingSink.end(SortedOps.java:357)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:485)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
        at org.openqa.grid.internal.NewSessionRequestQueue.processQueue(NewSessionRequestQueue.java:87)
        at org.openqa.grid.internal.DefaultGridRegistry.assignRequestToProxy(DefaultGridRegistry.java:236)
        at org.openqa.grid.internal.DefaultGridRegistry.access$200(DefaultGridRegistry.java:44)
        at org.openqa.grid.internal.DefaultGridRegistry$Matcher.run(DefaultGridRegistry.java:192)
21:11:34.528 WARN [MobileRemoteProxy.getNewSession] - 00008020-001824313CD2002E is not ready for a session. /status-wda error: null
```

**Comment by vdelendik** [comment]
fixed, mcloud:2.2 republished

**Comment by dhreben** [comment]
Verified on prod
